### PR TITLE
Better handling of http(s).proxyUser and http(s).proxyPassword

### PR DESCRIPTION
**Pull request by jborgland**
Suggested fix for HTTPCLIENT-1955.

### CODE REVIEW

File: httpclient/src/main/java/org/apache/http/impl/client/SystemDefaultCredentialsProvider.java [hunk 1]
```diff
@@ -116,25 +116,16 @@ public Credentials getCredentials(final AuthScope authscope) {
...
-            }
+			if (systemcreds == null) {
+				// Look for values given using http.proxyUser/http.proxyPassword or
+				// https.proxyUser/https.proxyPassword. We cannot simply use the protocol from
+				// the origin since a proxy retrieved from https.proxyHost/https.proxyPort will
+				// still use http as protocol
+				systemcreds = getProxyCredentials("http", authscope);
+				if (systemcreds == null) {
+					systemcreds = getProxyCredentials("https", authscope);
+				}
```
Comment (michael-o) [inline httpclient/src/main/java/org/apache/http/impl/client/SystemDefaultCredentialsProvider.java]: I haven't found any better solution yet, but this looks really redudant in most cases. Shouldn't it fall back for `https` => `http` only?

Comment (jborgland) [reply httpclient/src/main/java/org/apache/http/impl/client/SystemDefaultCredentialsProvider.java]: You mean to only do it if `protocol` is `http`? Sure. 

One could imagine other completely different types of fixes as well. I tried to make a fix that was completely local to the SystemDefaultCredentialsProvider but when it's called (from AuthenticationStrategyImpl) there's more information available - like the fact that it is in fact a proxy that we're connecting to, and the HTTP status code (one could for example decide to use these system properties as the primary source if the status code is 407).

**Comment by ok2c** [comment]
@jborgland The proposed change-set loos reasonable to me. Please _do_ fix the style check violations and make sure the build is successful and all tests pass.

File: httpclient/src/main/java/org/apache/http/impl/client/SystemDefaultCredentialsProvider.java (hunk 1 above)
Comment (ok2c) [reply httpclient/src/main/java/org/apache/http/impl/client/SystemDefaultCredentialsProvider.java]: @jborgland Feel free to propose API changes in master

**Comment by ok2c** [comment]
@jborgland I will commit the change-set to 4.5.x and master as soon as HC 5.0-beta3 release has been closed.

**Comment by jborgland** [comment]
@ok2c thank you! Sorry about the Checkstyle warnings, I was in a bit of hurry and didn't pay attention to Checkstyle not working in my (brand new) environment.

**Comment by ok2c** [comment]
Merged to master and 4.5.x as eac6c062b586196d32b7770d7052148acaf3c276 and 2ad70741e69e962bac41c57c2a5156dbc20e3e8d.

**Pull request closed without merging**